packages/oraiswap-v3/src/helpers.ts

```diff
@@ -538,8 +560,17 @@ export const calculateMinimumLiquidity = (
   lowerTick: number,
   upperTick: number,
   sqrtPrice: bigint,
-  slippage: number
+  slippage: number,
+  buildZapInMessageOptions?: buildZapInMessageOptions
 ) => {
+  if (buildZapInMessageOptions) {
+    const res = buildZapInMessageOptions.isTokenX
+      ? getLiquidityByX(BigInt(actualAmountXReceived.returnAmount), lowerTick, upperTick, sqrtPrice, true)
+      : getLiquidityByY(BigInt(actualAmountYReceived.returnAmount), lowerTick, upperTick, sqrtPrice, true);
+    message.minimumLiquidity = res.l ? (BigInt(res.l) * BigInt(100 - slippage)) / 100n : 0n;
```

Comment:
should use Decimal or other BigInt library that support floating point to avoid rounding